recipes/redboltz-mqtt_cpp/all/conanfile.py

```diff
@@ -1,63 +1,115 @@
+        "with_websocket": [True, False],
+        "mqtt_always_send_reason_code": [True, False],
+        "utf8_string": [True, False],
+        "with_logs": [True, False],
+    }
+    default_options = {
+        "cpp17":  False,
+        "fPIC":  True,
+        "with_static_boost":  False,
+        "with_static_openssl":  False,
+        "with_tls":  False,
+        "with_websocket":  False,
+        "mqtt_always_send_reason_code":  True,
+        "utf8_string":  False,
+        "with_logs": False
+    }
+
+    def source(self):
+        files.get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
 
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _min_cppstd(self):
+        return 11
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
 
     def requirements(self):
-        self.requires("boost/1.79.0")
+        self.requires("boost/1.80.0")
+        if self.options.with_tls or self.options.with_websocket:
+            self.requires("openssl/1.1.1s")
+            self.requires("zlib/1.2.13")
 
     def package_id(self):
         self.info.header_only()
 
-    @property
-    def _compilers_minimum_version(self):
-        return {
-            "gcc": "6",
-            "Visual Studio": "15.0",
-            "clang": "5",
-            "apple-clang": "10",
-        }
-    
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
     def validate(self):
-        if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, 14)
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, 11)
+            if self.options.get_safe("cpp17"):
+                check_min_cppstd(self, 17)
 
-        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
-        if minimum_version:
-            if tools.Version(self.settings.compiler.version) < minimum_version:
-                raise ConanInvalidConfiguration("{} requires C++14, which your compiler does not support.".format(self.name))
-        else:
-            self.output.warn("{} requires C++14. Your compiler is unknown. Assuming it supports C++14.".format(self.name))
-            
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["MQTT_BUILD_EXAMPLES"] = False
+        tc.variables["MQTT_BUILD_TESTS"] = False
+        tc.variables["MQTT_ALWAYS_SEND_REASON_CODE"] = self.options.mqtt_always_send_reason_code
+        tc.variables["MQTT_USE_STATIC_BOOST"] = self.options.with_static_boost
+        if self.options.with_static_boost:
+            openssl = self.dependencies["boost"]
+            openssl.shared = True
```

Comment:
```suggestion
            openssl = self.dependencies["boost"]
            openssl.shared = True
```
It's read-only, you can't force it. Please, remove it.